components/camera/videosource/webcam.go

```diff
@@ -257,6 +257,21 @@ func tryWebcamOpen(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
+
+	if attrs.Width != 0 && attrs.Height != 0 {
+		img, release, err := gostream.ReadMedia(ctx, source)
+		if release != nil {
+			defer release()
+		}
+		if err != nil {
+			return nil, err
+		}
+		if img.Bounds().Dx() != attrs.Width || img.Bounds().Dy() != attrs.Height {
+			return nil, errors.Errorf("requested width and height (%dx%d) are not available for this webcam"+
```

Comment:
[nit] add space before the rest of the error message

Reply:
done